controllers/operandrequest/reconcile_operand.go

```diff
@@ -1057,61 +1057,66 @@ func (r *Reconciler) updateK8sResource(ctx context.Context, existingK8sRes unstr
 			return true, nil
 		}
 
-		// isEqual := r.CheckAnnotation(existingK8sRes, newAnnotations) && r.CheckLabel(existingK8sRes, newLabels)
 		if k8sResConfig != nil {
+
 			k8sResConfigDecoded := make(map[string]interface{})
 			k8sResConfigUnmarshalErr := json.Unmarshal(k8sResConfig.Raw, &k8sResConfigDecoded)
 			if k8sResConfigUnmarshalErr != nil {
 				klog.Errorf("failed to unmarshal k8s Resource Config: %v", k8sResConfigUnmarshalErr)
 			}
 
-			for k, v := range k8sResConfigDecoded {
-				// isEqual = isEqual && reflect.DeepEqual(existingK8sRes.Object[k], v)
-				existingK8sRes.Object[k] = v
+			// Convert existing k8s resource spec to string
+			existingK8sResRaw, err := json.Marshal(existingK8sRes.Object)
+			if err != nil {
+				klog.Error(err)
+				return false, err
 			}
-		}
 
-		CRgeneration := existingK8sRes.GetGeneration()
+			// Merge spec from existing CR and OperandConfig spec
+			updatedExistingK8sRes := util.MergeCR(existingK8sResRaw, k8sResConfig.Raw)
 
-		// if isEqual {
-		// 	return true, nil
-		// }
+			r.EnsureAnnotation(existingK8sRes, newAnnotations)
+			r.EnsureLabel(existingK8sRes, newLabels)
+			if err := r.setOwnerReferences(ctx, &existingK8sRes, ownerReferences); err != nil {
+				return false, errors.Wrapf(err, "failed to set ownerReferences for k8s resource -- Kind: %s, NamespacedName: %s/%s", kind, namespace, name)
+			}
 
-		r.EnsureAnnotation(existingK8sRes, newAnnotations)
-		r.EnsureLabel(existingK8sRes, newLabels)
-		if err := r.setOwnerReferences(ctx, &existingK8sRes, ownerReferences); err != nil {
-			return false, errors.Wrapf(err, "failed to set ownerReferences for k8s resource -- Kind: %s, NamespacedName: %s/%s", kind, namespace, name)
-		}
+			if reflect.DeepEqual(existingK8sRes.Object, updatedExistingK8sRes) {
+				return true, nil
+			}
+			klog.Infof("updating k8s resource with apiversion: %s, kind: %s, %s/%s", apiversion, kind, namespace, name)
 
-		klog.V(2).Infof("updating k8s resource with apiversion: %s, kind: %s, %s/%s", apiversion, kind, namespace, name)
+			existingK8sRes.Object = updatedExistingK8sRes
```

Comment:
It seems that we are not `ensureAnnotation`, `ensureLabel` and `setOwnerReferences` on `updatedObject`. Currently it is applying to `existingK8sRes`

We might need to change the sequence here
1. deep copy a original version of `existingK8sRes.Object` - does not exists now.
2. `existingK8sRes.Object = updatedExistingK8sRes`
3. Ensure annotation, label and OwnerReference on new `updatedObject`
4. Check if new assigned object is deep equal with original one.